modules/notifications/notifications.php

```diff
@@ -727,69 +727,92 @@ function schedule_emails( $recipients, $subject, $message, $message_headers = ''
+					if ( $this->user_can_be_notified( $usergroup_user, $post_id ) ) {
+						$usergroup_recipients[] = $usergroup_user->user_email;
+					}
 				}
 			}
 		}
-		
-		$users = $this->get_following_users( $post_id, 'user_email' );
-		
-		// Merge arrays and filter any duplicates
-		$recipients = array_merge( $authors, $admins, $users, $usergroup_users );
-		$recipients = array_unique( $recipients );
-
-		// Process the recipients for this email to be sent
-		foreach( $recipients as $key => $user_email ) {
-			// Get rid of empty email entries
-			if ( empty( $recipients[$key] ) )
-				unset( $recipients[$key] );
-			// Don't send the email to the current user unless we've explicitly indicated they should receive it
-			if ( false === apply_filters( 'ef_notification_email_current_user', false ) && wp_get_current_user()->user_email == $user_email )
-				unset( $recipients[$key] );
+
+		$user_recipients = $this->get_following_users( $post_id, 'user_email' );
+		foreach( (array) $user_recipients as $key => $user ) {
+			$user_object = get_user_by( 'email', $user );
+			if ( ! $this->user_can_be_notified( $user_object, $post_id ) ) {
+				unset( $user_recipients[ $key ] );
+			}
 		}
-		
-		// Filter to allow further modification of recipients
+
+		// Merge arrays, filter any duplicates, and remove empty entries.
+		$recipients = array_filter( array_unique( array_merge( $admins, $user_recipients, $usergroup_recipients ) ) );
+
+		// Process the recipients for this email to be sent.
+		foreach(  $recipients as $key => $user_email ) {
+			// Don't send the email to the current user unless we've explicitly indicated they should receive it.
+			if ( false === apply_filters( 'ef_notification_email_current_user', false ) && wp_get_current_user()->user_email == $user_email ) {
+				unset( $recipients[ $key ] );
+			}
+		}
+
+		// Filter to allow further modification of recipients.
 		$recipients = apply_filters( 'ef_notification_recipients', $recipients, $post, $string );
-		
-		// If string set to true, return comma-delimited
+
+		// If string set to true, return comma-delimited.
 		if ( $string && is_array( $recipients ) ) {
 			return implode( ',', $recipients );
 		} else {
 			return $recipients;
 		}
 	}
-	
+
+	/**
+	 * Check if a user can be notified.
+	 * This is based off of the ability to edit the post/page by default.
+	 * 
+	 * @since 0.8.3
+	 * @param WP_User $user
+	 * @param int $post_id
+	 * @return bool True if the user can be notified, false otherwise.
+	 */
+	function user_can_be_notified( $user, $post_id ) {
+		$can_be_notified = false;
+
+		if ( $user instanceof WP_User && is_user_member_of_blog( $user->ID ) && is_numeric( $post_id ) ) {
+			// The 'edit_post' cap check also covers the undocumented 'edit_page' cap.
+			$can_be_notified = $user->has_cap( 'edit_post', $post_id );
+		}
+
+		return apply_filters( 'ef_notification_user_can_be_notified', $can_be_notified, $user, $post_id );
```

Comment:
There is a catch with the `$user` as it might be boolean `false` under certain circumstances ( `get_user_by` call in `_get_notification_recipients` might return `false` ). Thus, I would propose to document the filter via PHPDoc notifying the users about that possible issue.